File: RomanToInteger.cpp
class Solution {
public:

    int romanToInt(string s) {
        int result= 0;
        if(s.length() == 0)
        {
            return result;
        }
        
        for(int i=s.length()-1; i>=0; i--)
        {
            switch(s[i])
            {
                    case('I') :
                        if(s[i+1] && s[i+1] == 'V')
                        {
                            result -= 1;
                        }
                        else if(s[i+1] && s[i+1] == 'X')
                        {
                            result -= 1;
                        }
                        else
                        {
                            result += 1;
                        }
                        break;
                    
                    case('V') :
                        result+=5;
                        break;
                    
                    case('X') :
                        if(s[i+1] && s[i+1] == 'L')
                        {
                            result -= 10;
                        }
                        else if(s[i+1] && s[i+1] == 'C')
                        {
                            result -= 10;
                        }
                        else
                        {
                            result += 10;
                        }
                        break;
                    
                    case('L') :
                        result += 50;
                        break;
                    
                    case('C') :
                        if(s[i+1] && s[i+1] == 'D')
                        {
                            result -= 100;
                        }
                        else if(s[i+1] && s[i+1] == 'M')
                        {
                            result -= 100;
                        }
                        else
                        {
                            result += 100;
                        }
                        break;
                    
                    case('D') :
                        result += 500;
                        break;
                    
                    case('M') :
                        result += 1000;
                        break;
            }
        }
        return result;
    }
};

File: LearningPython3/RomanToInteger.py
class Solution:
    def romanToInt(self, s: str) -> int:
        counter = 0
        overflow = False
        for count, value in enumerate(s):
            
            if count+1 >= len(s):
                overflow = True
                
            if value is 'M':
                counter = counter + 1000
        
            if value is 'D':
                counter = counter + 500
            
            if value is 'C':
                if overflow is False and (s[count+1] is 'D' or s[count+1] is 'M'):
                    counter = counter - 100
                else:
                    counter = counter + 100
            
            if value is 'L':
                counter = counter + 50
            
            if value is 'X':
                if overflow is False and (s[count+1] is 'L' or s[count+1] is 'C'):
                    counter = counter - 10
                else:
                    counter = counter + 10
            
            if value is 'V':
                counter = counter + 5
            
            if value is 'I':
                if overflow is False and (s[count+1] is 'X' or s[count+1] is 'V'):
                    counter = counter -1
                else:
                    counter = counter + 1
                
        return counter
